[BUG] unit test fails when local kube cluster can not be touched

**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

When local file `.kube/config` exists but can not connect to kube cluster, it fails when execute `make test`.
```
I0301 12:06:45.277062   61566 replicas_test.go:428] test
--- FAIL: TestShutdown (30.01s)
    shutdown_engine_test.go:202: fail to call the shutdown with the error execute command [alluxio fsadmin report summary] with expectedErr: error sending request: Post "https://192.168.124.35:6443/api/v1/namespaces/fluid/pods/spark-master-0/exec?command=alluxio&command=fsadmin&command=report&command=summary&container=alluxio-master&container=alluxio-master&stderr=true&stdout=true": read tcp 198.18.0.1:49790->192.168.124.35:6443: read: connection reset by peer stdout  and stderr 
    shutdown_engine_test.go:202: fail to call the shutdown with the error execute command [alluxio fsadmin report summary] with expectedErr: error sending request: Post "https://192.168.124.35:6443/api/v1/namespaces/fluid/pods/hadoop-master-0/exec?command=alluxio&command=fsadmin&command=report&command=summary&container=alluxio-master&container=alluxio-master&stderr=true&stdout=true": read tcp 198.18.0.1:49892->192.168.124.35:6443: read: connection reset by peer stdout  and stderr 
FAIL
```

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

--- a/go.mod
+++ b/go.mod
@@ -55,7 +55,7 @@ replace k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.23.0
 replace k8s.io/sample-controller => k8s.io/sample-controller v0.23.0
 
 require (
-	github.com/agiledragon/gomonkey/v2 v2.2.0
+	github.com/agiledragon/gomonkey/v2 v2.9.0
 	github.com/brahma-adshonor/gohook v1.1.9
 	github.com/container-storage-interface/spec v1.7.0
 	github.com/docker/go-units v0.5.0

--- a/pkg/ddc/alluxio/shutdown_engine_test.go
+++ b/pkg/ddc/alluxio/shutdown_engine_test.go
@@ -1,9 +1,9 @@
 package alluxio
 
 import (
-	"strings"
 	"testing"
 
+	. "github.com/agiledragon/gomonkey/v2"
 	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
 	"github.com/fluid-cloudnative/fluid/pkg/ctrl"
 	"github.com/fluid-cloudnative/fluid/pkg/ddc/base"
@@ -194,12 +194,37 @@ func TestShutdown(t *testing.T) {
 		engine.name = test.runtimeInfo.GetName()
 		engine.namespace = test.runtimeInfo.GetNamespace()
 		engine.Helper = ctrl.BuildHelper(engine.runtimeInfo, client, engine.Log)
-		err := engine.Shutdown()
-		if !(err == nil ||
-			strings.Contains(err.Error(), "executable file not found") ||
-			strings.Contains(err.Error(), "invalid configuration") ||
-			strings.Contains(err.Error(), "not found")) {
-			t.Errorf("fail to call the shutdown with the error %v", err)
-		}
+
+		queryCacheStatusPatches := ApplyPrivateMethod(engine, "queryCacheStatus", func() (cacheStates, error) {
+			return cacheStates{
+				cacheCapacity:    "19.07MiB",
+				cached:           "0.00B",
+				cachedPercentage: "0.0%",
+				cacheHitStates: cacheHitStates{
+					bytesReadLocal:  20310917,
+					bytesReadUfsAll: 32243712,
+				},
+			}, nil
+		})
+
+		invokeCleanCachePatches := ApplyPrivateMethod(engine, "invokeCleanCache", func(path string) error {
+			return nil
+		})
+
+		func() {
+			defer invokeCleanCachePatches.Reset()
+			defer queryCacheStatusPatches.Reset()
+			err := engine.Shutdown()
+			if err != nil {
+				t.Fatalf("failed to engine.Shutdonw() due to: %v", err)
+			}
+		}()
+
+		// if !(err == nil ||
+		// 	strings.Contains(err.Error(), "executable file not found") ||
+		// 	strings.Contains(err.Error(), "invalid configuration") ||
+		// 	strings.Contains(err.Error(), "not found")) {
+		// 	t.Errorf("fail to call the shutdown with the error %v", err)
+		// }
 	}
 }

--- a/pkg/ddc/alluxio/shutdown_engine_test.go
+++ b/pkg/ddc/alluxio/shutdown_engine_test.go
@@ -211,7 +211,12 @@ func TestShutdown(t *testing.T) {
 			return nil
 		})
 
+		destroyMasterPatches := ApplyPrivateMethod(engine, "destroyMaster", func() error {
+			return nil
+		})
+
 		func() {
+			defer destroyMasterPatches.Reset()
 			defer invokeCleanCachePatches.Reset()
 			defer queryCacheStatusPatches.Reset()
 			err := engine.Shutdown()

